### PR TITLE
 Cognitive Services 2022-12-01

### PR DESCRIPTION
**Pull request by yangyuan**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
2022-12-01
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
2022-12-01
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by visingla-ms** [comment]
>             "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"

Use v4 types now.

Please apply to all the new APIs here.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:1603 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             }

Please don't add schema to 202 response. 202 should only return location header. Only 200 response should have response schema.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:1625 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "description": "If resource is created successfully, the service should return 201 (OK).",

Description needs updates, it should be specific to Patch/Update. Currently it is related to Put operation.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:1622 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "description": "If resource is created successfully or already existed, the service should return 200 (OK).",

Description needs updates, it should be specific to Patch/Update. Currently it is related to Put operation.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:1611 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "description": "The resource provider should return 200 (OK) to indicate that the operation completed successfully.\nFor a detailed explanation of each field in the response body, please refer to the request body description in the PUT resource section. The only GET specific properties are \"name,\" \"type\" and \"id.\"\nField\tDescription\nKind\trequired.  String.\n\tThe API set for this API account. It can be \n\t· a single API, for example: Face API, Vision API, Speech API. \n\ta bundle of APIs:  Face + Speech, Vision + Emotion, etc.\nsku.name\tRequired.\n\tThe pricing tier/plan of this API. Could be: \n\tF0 - Free\n\tB0 - Basic\n\tS0 - Standard\n\tP0 - Premium\n",

You can use description, specific to your scenario. This description uses the RPC guidance, whereas your customers will read this description for your APIs.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:1718 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "type": "array",

You should "m-ms-identifiers" for the arrays. See: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-identifiers](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-identifiers). It actually helps in identifying what is the unique property of items in that array.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:2373 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "$ref": "#/definitions/CommitmentPlanAccountAssociationProperties",

Use "x-ms-client-flatten" to avoid undesired nesting of properties. Please evaluate. See: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-client-flatten](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-client-flatten).

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:2396 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
Also, please check the linter issues mentioned in PR Report.

**Comment by yangyuan** [comment]
@visingla-ms 

About lint issue, as I explained in earlier comment and teams, we didn't touch that resource type (it's still same as previous version) it must be an issue in LintDiff tool.

**Comment by visingla-ms** [comment]
> @visingla-ms
> 
> About lint issue, as I explained in earlier comment and teams, we didn't touch that resource type (it's still same as previous version) it must be an issue in LintDiff tool.

In such a case, follow the guidelines for linter issues suppression: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process.

**Comment by yangyuan** [comment]
I addressed all other feedbacks except:

- Use v4 instead of v2. It seems that cannot mix use v2 and v4 for the same type.
- LintDiff suppress, which I assume will need time to approval.

@visingla-ms can you help to confirm that there is no other issues?

**Comment by visingla-ms** [comment]
>             "description": "If resource is updated successfully, the service should return 201 (OK)."

The description doesn't match, it says 201 (OK), however the response is 202.

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:1622 in b0a4c7b. [](commit_id = b0a4c7b35ed98b6082e1d4454e027084b0d13efc, deletion_comment = False)

**Comment by visingla-ms** [comment]
>             "description": "If resource is created successfully or already existed, the service should return 200 (OK).",

This needs to be fixed here.

---
In reply to: [1335859479](https://github.com/Azure/azure-rest-api-specs/pull/21716#issuecomment-1335859479) [](http://example.com/codeflow?ancestors=1335859479,1335859479)

---
Refers to: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2022-12-01/cognitiveservices.json:1611 in 9854192. [](commit_id = 9854192bab787b68d02739c361aa1687969a0b96, deletion_comment = False)

**Comment by visingla-ms** [comment]
> I addressed all other feedbacks except:
> 
> * Use v4 instead of v2. It seems that cannot mix use v2 and v4 for the same type.
> * LintDiff suppress, which I assume will need time to approval.
> 
> @visingla-ms can you help to confirm that there is no other issues?

There is one issue, where description of patch 200 and 202 response is not correct. See above review comments. Update these response descriptions.

Rest LGTM! SigningOff.

**Comment by xiaoxuqi-ms** [comment]
@yangyuan please fix the linter errors.

**Comment by yangyuan** [comment]
@xiaoxuqi-ms I believe the linter error is a false alert, as it says "The errors/warnings are introduced by current PR", but that resource type is already there before this PR.

**Comment by yangyuan** [comment]
Suppression request sent via https://aka.ms/SwaggerSuppressionRequest a week ago but didn't get an update ever since.

**Comment by xiaoxuqi-ms** [comment]
@jianyexi please  help to review the suppression request.

File: specification/cognitiveservices/resource-manager/readme.md [hunk 1]
```diff
@@ -87,6 +96,15 @@ input-file:
 - Microsoft.CognitiveServices/preview/2016-02-01-preview/cognitiveservices.json
 ```
 
+## Suppression
+``` yaml
+directive:
+  - suppress: TrackedResourcePatchOperation
+    from: cognitiveservices.json
+    where: $..paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/commitmentPlans/{commitmentPlanName}"]
```
Comment (jianyexi) [inline specification/cognitiveservices/resource-manager/readme.md:104]:
```suggestion
    where: $.paths["/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CognitiveServices/accounts/{accountName}/commitmentPlans/{commitmentPlanName}"]
```

Comment (jianyexi) [reply specification/cognitiveservices/resource-manager/readme.md:104]: why the resource has 'location' property if it's not tracked resource ?

**Review by xiaoxuqi-ms — APPROVED** [review]

**Pull request merged**